common:
  lid_topic: "/livox/lidar"
  imu_topic: "/livox/imu"
  time_sync_en: false         # ONLY turn on when external time synchronization is really not possible

preprocess:
  lidar_type: 1               # 1 for Livox serials LiDAR, 2 for Velodyne LiDAR, 3 for ouster LiDAR,
  scan_line: 4
  blind2: 1                   # 3.2 * 3.2    Adjust according to actual situation.
  max_range2: 1600            # 20*20 = 400 
  time_scale: 1e-3

mapping:
  acc_cov: 0.1
  gyr_cov: 0.1
  b_acc_cov: 0.0001
  b_gyr_cov: 0.0001
  fov_degree: 360
  det_range: 40
  extrinsic_est_en: false      # true: enable the online estimation of IMU-LiDAR extrinsic
  extrinsic_T: [ 0, 0, 0 ]
  extrinsic_R: [ 1, 0, 0,
                 0, 1, 0,
                 0, 0, 1 ]

publish:
  path_publish_en: true
  scan_publish_en: true         # false: close all the point cloud output
  scan_effect_pub_en: true      # true: publish the pointscloud of effect point
  dense_publish_en: false       # false: low down the points number in a global-frame point clouds scan.
  scan_bodyframe_pub_en: true   # true: output the point cloud scans in IMU-body-frame
  keyframe_pub_en: true

path_save_en: false 

pcd_save:
  pcd_save_en: true
  interval: -1                 # how many LiDAR frames saved in each pcd file;
  # -1 : all frames will be saved in ONE pcd file, may lead to memory crash when having too much frames.

feature_extract_enable: false
point_filter_num: 1
max_iteration: 3
filter_size_surf: 0.5
filter_size_map: 0.5 
cube_side_length: 1000

ivox_grid_resolution: 0.5        # default=0.2
ivox_nearby_type: 18             # 6, 18, 26
esti_plane_threshold: 0.1        # default=0.1
ivox_capacity: 1000000

location_mode: false
split_map : true
sub_grid_resolution: 80
load_g_map: "GlobalMap.pcd"
load_f_map: "FeatureMap.pcd"
load_eaf_size: 0.5
init_trans: [0, 0, 0]            # Initial value for repositioning
init_rpy: [0, 0, 0]              # degree. For example, 180 degrees

TL2C_tran: [0.00554806, -0.00113262, 0.0714708]
TL2C_rot: [0.928367, 0.00192075, 0.371661, 
          -0.00239883, 0.999997, 0.00082402, 
          -0.371658, -0.00165654, 0.928368]


